sn_cli/src/subcommands/files/mod.rs

```diff
@@ -231,11 +227,15 @@ async fn upload_files(
     // keep track of the progress in a separate task
     let progress_bar_clone = progress_bar.clone();
     let total_existing_chunks_clone = total_existing_chunks.clone();
-    tokio::spawn(async move {
+
+    let progress_handler = tokio::spawn(async move {
+        let mut upload_terminated_with_error = false;
+        // The channel is closes if the upload completes/errors out, so the loop is guaranteed to end.
         while let Some(event) = upload_event_rx.recv().await {
             match event {
-                FileUploadEvent::Uploaded(_addr) => {
+                FileUploadEvent::Uploaded(addr) => {
                     progress_bar_clone.inc(1);
+                    chunk_manager.mark_completed(std::iter::once(*addr.xorname()));
```

Comment:
`FileUploadEvent::Uploaded` only says the chunk got sent out to the network.
But without a `fetch verification`, can't say the `chunk is stored in the network`.
Hence, marking it as `completed` here is bit risky, and not consistent with the current definition of `completed` (verified to be existing in the network, when the verification flag is set) ?

Reply:
> `FileUploadEvent::Uploaded` only says the chunk got sent out to the network.

This event is emitted when the `Files::upload_chunk(verify_store,...)` returns `Ok(())`.  Thus, if we have configured the `Files::set_verify_store(true)`, we should indeed be verifying the upload.